modules/test/playwright/tests/frontend-data-set-views-web/selectionFilters.spec.ts

```diff
@@ -99,3 +102,160 @@ test.describe('Filters in the Data Set Manager', () => {
+			});
+
+			await fragmentTest.step(
+				'Add a field, so FDS has something to show',
+				async () => {
+					await dataSetManagerApiHelpers.createDataSetField({
+						label_i18n: {en_US: 'Renderer'},
+						name: 'renderer',
+						r_fdsViewFDSFieldRelationship_c_fdsViewERC:
+							filtersDataSetViewERC,
+					});
+
+					await dataSetManagerApiHelpers.createDataSetField({
+						label_i18n: {en_US: 'Sortable'},
+						name: 'sortable',
+						r_fdsViewFDSFieldRelationship_c_fdsViewERC:
+							filtersDataSetViewERC,
+						renderer: 'boolean',
+					});
+				}
+			);
+
+			await fragmentTest.step(
+				'Create a new selection filter',
+				async () => {
+					const picklist = await picklistApiHelpers.getPicklist(
+						picklistName
+					);
+
+					await dataSetManagerApiHelpers.createDataSetSelectionFilter(
+						{
+							fieldName: 'renderer',
+							label_i18n: {en_US: filterLabel},
+							r_fdsViewFDSDynamicFilterRelationship_c_fdsViewERC:
+								filtersDataSetViewERC,
+							source: picklist.externalReferenceCode,
+							sourceType: 'PICKLIST',
+						}
+					);
+				}
+			);
+
+			await fragmentTest.step('Configure Data Set fragment', async () => {
+				await fdsFragmentPage.configureDataSetFragment({
+					layout,
+					viewLabel: filtersDataSetViewLabel,
+				});
+			});
+
+			await fragmentTest.step(
+				'Check current items in the Frontend Data Set',
+				async () => {
+					await expect(
+						fdsFragmentPage.page.getByText(
+							'Showing 1 to 2 of 2 entries.'
+						)
+					).toBeVisible();
+				}
+			);
+
+			await fragmentTest.step(
+				'Filters are available in the fragment',
+				async () => {
+					await expect(
+						fdsFragmentPage.page.getByRole('button', {
+							name: 'Filter',
+						})
+					).toBeVisible();
+				}
+			);
+
+			await fragmentTest.step('Open filters component', async () => {
+				await fdsFragmentPage.page
+					.getByRole('button', {name: 'Filter'})
+					.click();
+			});
+
+			await fragmentTest.step('Select filter', async () => {
+				await expect(
+					fdsFragmentPage.page.getByRole('menuitem', {
+						name: filterLabel,
+					})
+				).toBeVisible();
+				await fdsFragmentPage.page
+					.getByRole('menuitem', {name: filterLabel})
+					.click();
+				await expect(
+					fdsFragmentPage.page.getByRole('radio', {
+						name: picklistDefaultOption,
+					})
+				).toBeVisible();
+				await expect(
+					fdsFragmentPage.page.getByRole('radio', {
+						name: picklistBooleanOption,
+					})
+				).toBeVisible();
+
+				await fdsFragmentPage.page
+					.getByRole('radio', {name: picklistBooleanOption})
+					.check();
+				await fdsFragmentPage.page
+					.getByRole('button', {name: 'Add filter'})
+					.click();
+
+				// Close filter
+
+				await fdsFragmentPage.page.keyboard.press('Escape');
+			});
+
+			await fragmentTest.step('Check that the filter works', async () => {
+				await expect(
+					fdsFragmentPage.page.getByText(
+						'Showing 1 to 1 of 1 entries.'
+					)
```

Comment:
This is ok, but probably can be improved. This means that _some_ filtering happened, but not that it was correct. Can we check if specific entry is in UI, that matches the picklist option?